File: lib/src/data/model/event.dart
import 'package:meta/meta.dart';

@immutable
class Event {
  final String name;
  final Map<String, dynamic> properties;
  final DateTime? timestamp;

  const Event({
    required this.name,
    this.properties = const {},
    this.timestamp,
  });

  @override
  String toString() {
    return 'Event{'
        'name: $name, '
        'properties: $properties, '
        'timestamp: $timestamp'
        '}';
  }
}

File: android/src/main/kotlin/com/exponea/data/Event.kt
package com.exponea.data

data class Event(val name: String, val properties: Map<String, Any>, val timestamp: Double?) {
    companion object {
        fun fromMap(map: Map<String, Any?>): Event {
            return Event(
                name = map["name"] as String? ?: throw IllegalStateException("Event.name is required!"),
                properties = (map["properties"] as Map<String, Any>?) ?: mapOf(),
                timestamp = map["timestamp"] as Double?
            )
        }
    }
}
